packages/gatsby/src/datastore/in-memory/run-fast-filters.ts

```diff
@@ -406,9 +404,6 @@ function convertAndApplyFastFilters(
     if (stats) {
       stats.totalIndexHits++
     }
-    if (firstOnly) {
-      return result.slice(0, 1)
-    }
```

Comment:
Now the calling side must pass `limit: 1` (and empty `sort`) instead of `firstOnly`.